roles/ceph-config/tasks/main.yml

```diff
@@ -13,19 +13,22 @@
     set_fact:
       num_osds: "{{ devices | length }}"
     when:
-      - devices|length > 0 and (osd_scenario == 'collocated' or osd_scenario == 'non-collocated')
+      - devices|length > 0
+      - (osd_scenario == 'collocated' or osd_scenario == 'non-collocated')
 
   - name: count number of osds for lvm scenario
     set_fact:
       num_osds: "{{ lvm_volumes | length }}"
     when:
-      - lvm_volumes|length > 0 and osd_scenario == 'lvm'
+      - lvm_volumes|length > 0
+      - osd_scenario == 'lvm'
 
   - name: count number of osds for lvm-batch scenario
     set_fact:
       num_osds: "{{ (ceph-volume lvm batch --report devices | from_json()).osds | length }}"
     when:
-      - devices|length > 0 and osd_scenario == 'lvm-batch'
+      - devices|length > 0
+      - osd_scenario == 'lvm-batch'
```

Comment:
This needs to be ``- osd_scenario == 'lvm'``

Reply:
yeah, updating it in my next commit with the lvm-batch changes.